src/net.h

```diff
@@ -716,6 +752,25 @@ class CNode
         BOOST_FOREACH(CNetMessage &msg, vRecvMsg)
             msg.SetVersion(nVersionIn);
     }
+    void SetSendVersion(int nVersionIn)
+    {
+        // Send version may only be changed in the version message, and
+        // only one version message is allowed per session. We can therefore
+        // treat this value as const and even atomic as long as it's only used
+        // once the handshake is complete. Any attempt to set this twice is an
+        // error.
+        assert(nSendVersion == 0);
+        *const_cast<int*>(&nSendVersion) = nVersionIn;
```

Comment:
Ok, I'll just change this back to non-const. The const was only to prove that it's not racy, since it can only be set once.